googleapiclient/http.py

```diff
@@ -1120,10 +1120,6 @@ def _serialize_request(self, request):
     g.flatten(msg, unixfrom=False)
     body = fp.getvalue()
 
-    # Strip off the \n\n that the MIME lib tacks onto the end of the payload.
-    if request.body is None:
```

Comment:
I don't think I like the blind truncation here - is it guaranteed by the MIME library that the string it returns will end in two newlines, or is that an implementation detail? Let's be a little safer here and say

if request.body is None and body[-2:] == '\n\n':

? Why is

body = body.rstrip()

not the appropriate action to take here?